modules/tinydnsbackend/tinydnsbackend.cc

```diff
@@ -166,6 +166,8 @@ void TinyDNSBackend::getAllDomains(vector<DomainInfo> *domains, bool include_dis
 
   while (get(rr)) {
     if (rr.qtype.getCode() == QType::SOA) {
+      if (std::find_if(domains->begin(), domains->end(), [rr](const DomainInfo& d){ return rr.qname == d.zone; }) != domains->end())
```

Comment:
It looks like the complexity is terrible (O(n^2)), which could be an issue with very large datasets. Perhaps we could use an `unordered_set` instead?

Reply:
True.. it's possible, but would though mean that someone is keeping several thousand domains in tinydns. But sure, I'll change it to unordered_set instead.